test/test_amity.py

```diff
@@ -0,0 +1,415 @@
+        """
+        self.staff = staff.Staff("King", "Kong")
+        self.assertIsInstance(self.staff, person.Person)
+
+    def test_office_is_subclass_of_room(self):
+        """
+        Office class should subclass Room class
+        """
+        self.office = office.Office("Oculus")
+        self.assertIsInstance(self.office, room.Room)
+
+    def test_livingroom_subclasses_room(self):
+        """
+        LivingRoom class should subclass Room class
+        """
+        self.livroom = livingroom.LivingRoom("Bedroom")
+        self.assertIsInstance(self.livroom, room.Room)
+
+
+class TestRoomClass(unittest.TestCase):
+    """
+    Tests the functionality of Room class, and its subclasses LivingRoom and Office
+    """
+
+    def test_office_room_has_zero_occupants_by_default(self):
+        """
+        By default, a Office room created should have zero occupants
+        """
+        self.rm = office.Office("Hogwarts")
+        self.assertEqual(self.rm.get_occupants(), 0)
+
+    def test_get_name_office_room(self):
+        """
+        Should return the name of the office room, formatted in title case
+        """
+        self.rm = office.Office("Quiet room")
+        self.assertEqual(self.rm.get_name(), "Quiet Room")
+
+    def test_get_name_livingroom(self):
+        """
+        Should be able to get name of LivingRoom room, formatted in title case
+        """
+        self.lvroom = livingroom.LivingRoom("Chillout place")
+        self.assertEqual(self.lvroom.get_name(), "Chillout Place")
+
+    def test_can_add_fellow_to_office_room(self):
+        """
+        Should be able to add fellows to Office room
+        """
+        self.of = office.Office("New Office")
+        self.fel = fellow.Fellow("Mary", "Jane")
+        self.of.add_person(self.fel)
+        self.assertEqual(self.fel, self.of.occupants[0])
+
+    def test_can_add_fellow_to_livingroom_room(self):
+        """
+        Should be able to add fellow to LivingRoom rooms
+        """
+        self.lroom = livingroom.LivingRoom("Kitchen")
+        self.fellow = fellow.Fellow("May", "Teresa")
+        self.lroom.add_person(self.fellow)
+        self.assertEqual(self.fellow, self.lroom.occupants[0])
+
+    def test_can_add_staff_to_office_room(self):
+        """
+        Should be able to add staff to Office room
+        """
+        self.staff = staff.Staff("TT", "PP")
+        self.office = office.Office("Main Office")
+        self.office.add_person(self.staff)
+        self.assertEqual(self.staff, self.office.occupants[0])
+
+    def test_livingroom_raise_error_for_addition_of_staff(self):
+        """
+        LivingRoom should raise an InvalidRoomOccupantError exception when Staff tries to join it.
+        """
+        self.staff = staff.Staff("TT", "PP")
+        self.lroom = livingroom.LivingRoom("Livingroom")
+        with self.assertRaises(InvalidRoomOccupantError):
+            self.lroom.add_person(self.staff)
```

Comment:
With this approach, I wanted to check whether the exception is raised. But I see sense in checking for a particular return value.